.github/workflows/ci.yaml

```diff
@@ -5,9 +5,54 @@ on:
   pull_request:
     branches: [ "main" ]
 
+env:
+  DOCKER_BUILDX_PLATFORMS: linux/amd64,linux/arm64
+
 jobs:
+  # Make sure that Kubebuilder autogenerated files are up to date. 
+  kubebuilder-diff:
+    name: Kubebuilder Diff
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19 # Use the latest
+    - name: Generate manifests
+      run: make manifests
+    - name: Fail if there are uncommited manifest changes
+      run: |
+        git diff --exit-code
+        git diff --cached --exit-code
+
+  build-images:
+    name: Build Images
+    needs:
+    - kubebuilder-diff
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
```

Comment:
these are pretty arbitrary I assume. Are they just a good practice safeguard to prevent an action from running rogue and burning through your build minutes?

Reply:
Yep. I had a bad experience so now do it as a safeguard against accidents